### PR TITLE
Improve error msg when using bitsandbytes

### LINKED ISSUE #30751: BitsNBytes 4 bit quantization error message typo and logical errors in error message handling
### System Info

Newest version of transformers, accelerate, bitsandbytes in a docker container (nvidia/cuda:12.4.1-cudnn-runtime-ubuntu22.04), Ubuntu 20.04 system, Arch Linux laptop

### Who can help?

@SunMarc and @younesbelkada

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
from transformers import RobertaForSequenceClassification, RobertaTokenizer
eval_model_path = "hubert233/GPTFuzz"
tokenizer = RobertaTokenizer.from_pretrained(eval_model_path)
eval_model = RobertaForSequenceClassification.from_pretrained(
    eval_model_path,
    low_cpu_mem_usage=True,
    device_map="auto",
    quantization_config=BitsAndBytesConfig(
        load_in_4bit=True, bnb_4bit_compute_dtype=torch.float16
    ),
)
eval_model.eval()
```

Impacts other models when using BNB 4 bit, e.g. meta-llama/Llama-2-7b-chat-hf

### Expected behavior

On a system where cuda isn't working (e.g. a laptop with no Nvidia GPU or a container that wasn't properly connected to the host OS driver), running that code snippet gives an import error:

image

The logic at issue starts here:

https://github.com/huggingface/transformers/blob/e0c3cee17085914bbe505c159beeb8ae39bc37dd/src/transformers/quantizers/quantizer_bnb_4bit.py#L60

1. The `if not torch.cuda.is_available():` error message seemingly is frequently bypassed on multiple systems with no torch cuda functionality, passing the system state onto `if not (is_accelerate_available() and is_bitsandbytes_available()):`
2.  `is_bitsandbytes_available())` calls out to https://github.com/huggingface/transformers/blob/e0c3cee17085914bbe505c159beeb8ae39bc37dd/src/transformers/utils/import_utils.py#L749 and `is_accelerate_available()` calls out to https://github.com/huggingface/transformers/blob/e0c3cee17085914bbe505c159beeb8ae39bc37dd/src/transformers/utils/import_utils.py#L819 . In the instance where the first is check is bypassed, the `if not torch.cuda.is_available():` function that's built into the `is_bitsandbytes_available())` fails, resulting in getting a wildly confusing error message of "Using `bitsandbytes` 8-bit quantization requires Accelerate: `pip install accelerate` and the latest version of bitsandbytes: `pip install -i https://pypi.org/simple/ bitsandbytes`" when accelerate and bitsandbytes are both installed.
3. Ignoring the issue in the torch call that resulted in my getting into this in the first place, I believe that `and` is tthe incorrect logical operator to use in https://github.com/huggingface/transformers/blob/e0c3cee17085914bbe505c159beeb8ae39bc37dd/src/transformers/quantizers/quantizer_bnb_4bit.py#L63 , because the error message says the problem is in accelerate not being present while bitsandbytes is
4. The error message says "Using `bitsandbytes` 8-bit quantization" in the 4-bit version of the file
5. I _suspect_ that the original issue with the torch cuda availability check being erroneously passed is caused by the logic in this line: https://github.com/huggingface/transformers/blob/e0c3cee17085914bbe505c159beeb8ae39bc37dd/src/transformers/quantizers/quantizer_bnb_4bit.py#L29

Additionally, having `is_bitsandbytes_available())` call `torch.cuda.is_available()` seems like a non obvious and non modular design choice that's likely to result in similar very misleading and hard to debug error messages in the future.

I think that the underlying logical issues here also likely resulted in this other github sub-issue: https://github.com/huggingface/transformers/issues/29177#issuecomment-1957180468

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ?

This PR fixes a typo in an error message and make it easier to debug when getting an import error for bitsandbytes. 

Fixes https://github.com/huggingface/transformers/issues/30751

### CODE REVIEW

File: src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py
Comment (younesbelkada) [inline src/transformers/models/xlm_roberta_xl/modeling_xlm_roberta_xl.py:1]: Is this change intended?

File: tests/models/xlm_roberta_xl/test_modeling_xlm_roberta_xl.py
Comment (younesbelkada) [inline tests/models/xlm_roberta_xl/test_modeling_xlm_roberta_xl.py:1]: Is this change intended?

**Review by younesbelkada — COMMENTED** [review]
Thanks ! left two comments about potentially unneeded changes

**Comment by SunMarc** [comment]
yes mb i pushed the wrong branch. let me fix this !

**Review by younesbelkada — APPROVED** [review]
Thanks a lot !

**Review by amyeroberts — APPROVED** [review]
Thanks! 

Is this only difference between the 4bit and 8bit validation the required version? If so, we  could always abstract out a common set of things to validate and the just have `quantizer_bnb_4bit.validate_environment` use that + the version check

**Comment by SunMarc** [comment]
Makes sense ! I think that this was the goal of `required_packages` attribute in `HfQuantizer` but we never used it in the end. I will do that in a follow up PR for all quants methods.
